### DISCUSSION TITLE
addProperty alternative on IndexingConfigurationBuilder

### DISCUSSION THREAD

**Discussion by JM92103** [infinispan/infinispan#12552] [Q&A]
Hi, we are in the process of upgrading Quarkus to 3.9.5, which means upgrading Infinispan to 15.0.1.Final.
We are facing a small issue in that on our ConfigurationBuilder, we are adding a property of `hibernate.search.backend.analysis.configurer` to add custom analyzers to our indexes. This is now deprecated, and I am unable to find any alternative in reader() or writer(). 
Setting this in application.properties doesn't work. Is there any way to send properties directly to hibernate-search anymore?

**Comment by fax4ever** [comment]
Hi @JM92103!
Try to register your implementation of  `org.hibernate.search.backend.lucene.analysis.LuceneAnalysisConfigurer` as a Java standard service (using the same type `org.hibernate.search.backend.lucene.analysis.LuceneAnalysisConfigurer`.

**Reply by JM92103** [reply]
Hi @fax4ever, thank you for your reply. 
Please can you elaborate on what you mean? I have defined my Analyzer as follows:
```
@Dependent
@Named("fundDeskNameAnalyzer")
public class AnalyzerConfigurer implements LuceneAnalysisConfigurer {
    
    @Override
    public void configure(LuceneAnalysisConfigurationContext context) {
        context.analyzer("fundDeskNameAnalyzer").custom()
                .tokenizer(SimplePatternSplitTokenizerFactory.class).param(PATTERN,":")
                .tokenFilter(LowerCaseFilterFactory.class);

    }
}
```
You can see my attempts at turning it into a Bean by using @Dependent and @Named, but sadly with no luck in it being picked up.

**Reply by fax4ever** [reply]
Actually you should use the Java standard service, the one you should register in the:
META-INF/services/org.hibernate.search.backend.lucene.analysis.LuceneAnalysisConfigurer
